lightning/src/ln/msgs.rs

```diff
@@ -634,6 +637,43 @@ impl Readable for NetAddress {
 	}
 }
 
+impl FromStr for NetAddress {
+    type Err = AddrParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let host: String = s.split("-").collect::<Vec<&str>>()[0].parse().expect("Invalid hostname");
```

Comment:
I started with `:` but changed to `-` because IPv6 can have multiple `:`. I can still use `:` and slice by the last `:` occurrence or use `@` that is not too bad as well.

Reply:
I started with `:` but changed to `-` because IPv6 can have multiple `:`. I can still use `:` and slice by the last `:` occurrence or use `@` that is not too bad as well.